### PR TITLE
Rust eth

### PR DESCRIPTION
**Pull request by benma**

### CODE REVIEW

**Comment by NickeZ** [comment]
Can you create a new feature named "app-ethereum" for example here: https://github.com/digitalbitbox/bitbox02-firmware/blob/884dad28fd88844a557a575182741414370d8a99/src/rust/bitbox02-rust-c/Cargo.toml#L27

And then add a define for that feature here: https://github.com/digitalbitbox/bitbox02-firmware/blob/884dad28fd88844a557a575182741414370d8a99/src/rust/bitbox02-cbindgen.toml#L36

And then add `RUST_APP_ETHEREUM` (or what you named it) here: https://github.com/digitalbitbox/bitbox02-firmware/blob/884dad28fd88844a557a575182741414370d8a99/src/platform/platform_config.h#L58

**Comment by NickeZ** [comment]
And theeen, create a feature to collect the features for this product, example:

```
[features]
bitbox02-multi = ["platform-bitbox02", "app-ethereum"]
```

File: src/platform/platform_config.h [hunk 1]
```diff
@@ -57,6 +57,7 @@
 
 #if PRODUCT_BITBOX_MULTI == 1 && !defined(BOOTLOADER)
 #define RUST_BITBOX_MULTI
+#define RUST_BITBOX02
```
Comment (benma) [inline src/platform/platform_config.h]: i'll remove both in another PR and move them to cbindgen as discussed

File: src/rust/bitbox02-rust-c/src/app_ethereum.rs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#[no_mangle]
+pub extern "C" fn rust_ethereum_keypath_is_valid_address(
+    keypath: *const util::c_types::uint32_t,
```
Comment (NickeZ) [inline src/rust/bitbox02-rust-c/src/app_ethereum.rs]: Is there any benefit to taking `util::c_types::uint32_t` over `u32`?

**Pull request merged**